### PR TITLE
[Modules] Updated OperationalInsights/workspaces to new dependencies approach #2177

### LINKED ISSUE #2177: Update `OperationalInsights/workspaces` to new dependencies approach

### PR DESCRIPTION
**Pull request by elanzel**
# Description

Updated Log Analytics Workspace to new dependencies approach.
Close #2177 Update OperationalInsights/workspaces to new dependencies approach

| Pipeline |
| - |
|[OperationalInsights: Workspaces](https://github.com/Azure/ResourceModules/actions/workflows/ms.operationalinsights.workspaces.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

# Checklist

- [x] I'm sure there are no other open Pull Requests for the same update/change
- [x] My corresponding pipelines / checks run clean and green without any errors or warnings
- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation (readme)
- [x] I did format my code

### CODE REVIEW

File: modules/Microsoft.OperationalInsights/workspaces/.test/common/dependencies.bicep [hunk 1]
```diff
@@ -0,0 +1,33 @@
+@description('Optional. The location to deploy to.')
+param location string = resourceGroup().location
+
+@description('Required. The name of the Storage Account to create.')
+param storageAccountName string
+
+@description('Required. Name of the Automation Account to create.')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/common/dependencies.bicep:7]:
```suggestion
@description('Required. The name of the Automation Account to create.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/common/dependencies.bicep [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    sku: {
+      name: 'Basic'
+    }
+  }
+}
+
+@description('The resource ID of the created Storage Account.')
+output storageAccountResourceId string = storageAccount.id
+
+@description('The resource ID of the deployed automation account.')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/common/dependencies.bicep:32]:
```suggestion
@description('The resource ID of the created Automation Account.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,198 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,198 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.operationalinsights.workspaces-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,198 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/min/deploy.test.bicep [hunk 6]
```diff
@@ -0,0 +1,37 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/min/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/min/deploy.test.bicep [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+@maxLength(90)
+param resourceGroupName string = 'ms.operationalinsights.workspaces-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/min/deploy.test.bicep:10]:
```suggestion
@description('Optional. The location to deploy resources to.')
```

File: modules/Microsoft.OperationalInsights/workspaces/.test/min/deploy.test.bicep [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints')
```
Comment (eriqua) [inline modules/Microsoft.OperationalInsights/workspaces/.test/min/deploy.test.bicep:13]:
```suggestion
@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
```

**Comment by elanzel** [comment]
Thank you @eriqua , all comments are resolved and i've enabled the auto-merge.
Thank you for you support

File: modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep [hunk 9]
```diff
@@ -0,0 +1,198 @@
...
+  name: resourceGroupName
+  location: location
+}
+
+module resourceGroupResources 'dependencies.bicep' = {
+  scope: resourceGroup
+  name: '${uniqueString(deployment().name, location)}-paramNested'
+  params: {
+    storageAccountName: 'dep<<namePrefix>>sa${serviceShort}'
+    automationAccountName: 'dep<<namePrefix>>aa${serviceShort}'
```
Comment (MrMCake) [inline modules/Microsoft.OperationalInsights/workspaces/.test/common/deploy.test.bicep:32]:
```suggestion
    automationAccountName: 'dep-<<namePrefix>>-auto-${serviceShort}'
```

**Pull request merged**